### PR TITLE
Next

### PR DESCRIPTION
**Pull request by Mirodil**
for koa2

### CODE REVIEW

**Comment by shellscape** [comment]
thanks for the PR. just got back into town from the holidays 🎅 , will have a look tomorrow.

File: index.js [hunk 1]
```diff
@@ -18,12 +18,12 @@ var util = require('util');
 var merge = function (obj1, obj2) {
   var c = {};
   var keys = Object.keys(obj2);
-  for(var i=0; i!==keys.length; i++) {
+  for (var i = 0; i !== keys.length; i++) {
```
Comment (shellscape) [inline index.js]: please don't refactor spacing and style

File: index.js [hunk 2]
```diff
@@ -18,12 +18,12 @@ var util = require('util');
...
     c[keys[i]] = obj2[keys[i]];
   }
 
   keys = Object.keys(obj1);
-  for(i=0; i!==keys.length; i++) {
+  for (i = 0; i !== keys.length; i++) {
```
Comment (shellscape) [inline index.js]: please don't refactor spacing and style

File: index.js [hunk 3]
```diff
@@ -37,22 +37,26 @@ var merge = function (obj1, obj2) {
...
+  });
 };
 
 /**
  * @class MissingTemplateError
  * @param {String} message The error message
  * @param {Object} extra   The value of the template, relating to the error.
  */
-function MissingTemplateError (message, extra) {
+function MissingTemplateError(message, extra) {
```
Comment (shellscape) [inline index.js]: please don't refactor spacing and style

File: index.js [hunk 4]
```diff
@@ -66,7 +70,7 @@ util.inherits(MissingTemplateError, Error);
  * @param {String} message The error message
  * @param {Object} extra   Misc infomration.
  */
-function BadOptionsError (message, extra) {
+function BadOptionsError(message, extra) {
```
Comment (shellscape) [inline index.js]: please don't refactor spacing and style

File: index.js [hunk 5]
```diff
@@ -85,7 +89,7 @@ exports = module.exports = new Hbs();
  * expose method to create additional instances of `Hbs`
  */
 
-exports.create = function() {
+exports.create = function () {
```
Comment (shellscape) [inline index.js]: there's a bunch of other instances where it looks like your editor modified spacing. please clean those up

File: test/render.js [hunk 6]
```diff
@@ -231,23 +232,23 @@ describe('rendering', function() {
...
 });
 
 describe('var conflict', function () {
-  var app = koa();
+  var app = new koa();
   app.use(hbs.middleware({
     viewPath: __dirname + '/app/assets'
   }));
-  app.use(function * () {
+  app.use(function* () {
```
Comment (fiftyfactorauthentication) [inline test/render.js]: needs to be updated to use async

**Comment by fiftyfactorauthentication** [comment]
@Mirodil looks good, but needs the render tests to be updated for async. that's the only blocker on merging this.

**Comment by Mirodil** [comment]
tests already has been updated

File: test/render.js (hunk 6 above)
Comment (Mirodil) [inline test/render.js]: done

File: index.js (hunk 5 above)
Comment (Mirodil) [inline index.js]: done

**Comment by fiftyfactorauthentication** [comment]
@Mirodil why did you remove _all_ spaces? 

"please don't refactor spacing and style" only meant to not modify **_existing_** spacing/formatting/code style.

**Comment by shellscape** [comment]
@Mirodil ping

**Comment by Mirodil** [comment]
we can close this pull request

**Comment by shellscape** [comment]
ok will do. we'll cherry pick the changes and give you credit for them.

**Pull request closed without merging**